Add command to reconfigure keys

autrilla opened the pull request:
Currently there's no easy way to add or remove keys from files en masse. This adds a way to update the keys in all files in a directory (or also for a single file) using the `.sops.yaml` config file. It prompts the user to confirm the change to every file (unless the `-y` flag is provided), and shows a diff so the user can see what changes will be made.

jvehent commented on cmd/sops/main.go:
```diff
@@ -211,6 +213,33 @@ func main() {
+			Flags: append([]cli.Flag{
+				cli.StringFlag{
+					Name:  "file, f",
+					Usage: `the file or directory that should be updated with the new keys. If providing a directory, SOPS will walk it and update all files in all subdirectories. '.git' directories and '.sops.yaml' files are ignored.`,
```
Using a flag is different from the regular sops command line that just takes the file from the first non-flag argument. I think we should preserve it. 

As far as passing a whole directory, I wonder if that's needed, because ops can always use something along the line of `find . -name '*.yaml' -exec sops updatekeys {} \;`

jvehent commented on cmd/sops/main.go:
```diff
@@ -211,6 +213,33 @@ func main() {
+				},
+				cli.BoolFlag{
+					Name:  "yes, y",
+					Usage: `automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively.`,
```
`pre-approve all changes and run non-interactively`

jvehent commented on cmd/sops/subcommand/updatekeys/updatekeys.go:
```diff
@@ -0,0 +1,198 @@
+
+import (
+	"fmt"
+
```
nit: gofmt

autrilla replied:
Hmm, I don't know how this happens, but `goimports` doesn't seem to care about it. I'll fix it manually.

jvehent commented on cmd/sops/subcommand/updatekeys/updatekeys.go:
```diff
@@ -0,0 +1,198 @@
+				return nil
+			}
+			ignoredPaths := []string{
+				"/.git/",
```
As a general rule, I don't like hardcoding exceptions. I'm also not sure that's needed because `.git` store binary blobs, not sops file, and `.sops.yaml` doesn't have sops metadata.

autrilla replied:
The reasoning for this was to eliminate hundreds of error messages when running it, but if we go the route of just accepting a single file, it won't be needed at all.

jvehent commented on cmd/sops/subcommand/updatekeys/updatekeys.go:
```diff
@@ -0,0 +1,198 @@
+				".sops.yaml",
+			}
+			for _, ignoredPath := range ignoredPaths {
+				if strings.Contains(path, ignoredPath) {
```
this will trigger on `foo.sops.yaml` which isn't a sops configuration file

jvehent commented on cmd/sops/subcommand/updatekeys/updatekeys.go:
```diff
@@ -0,0 +1,198 @@
+	return nil
+}
+
+func syncFile(opts Opts) error {
```
`sync` defines a specific operation (move a state from a source to a destination) that is different from what you're doing here. I'd suggest `update` or `modify`.

autrilla commented:
All comments should be addressed now.

- Removed the "recursiveness".
- Updated the usage description of `-y`
- Use arguments instead of flags for the file to update
- Don't ignore files containing `.git` or `.sops.yaml`.
